### PR TITLE
Jdbc read partitioning

### PR DESCRIPTION
**Pull request by shashank-google**

### CODE REVIEW

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java [hunk 1]
```diff
@@ -65,17 +80,40 @@ public void runTemplate() {
               + "in resources/conf/template.properties file or at runtime. Refer to jdbc/README.md for more instructions.");
     }
 
+    if (StringUtils.isNotBlank(concatedPartitionProps)
```
Comment (snrssc) [inline java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java:83]: This logic makes parallel read as mandatory and job fails if partitioning is not specified.

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java [hunk 2]
```diff
@@ -65,17 +80,40 @@ public void runTemplate() {
...
+            + "5. {}:{}"
+            + "6. {}:{}"
+            + "7. {}:{}",
         JDBC_TO_GCS_OUTPUT_LOCATION,
-        GCSOutputLocation,
+        gcsOutputLocation,
         JDBC_TO_GCS_WRITE_MODE,
-        GCSWriteMode,
+        gcsWriteMode,
         JDBC_TO_GCS_JDBC_URL,
```
Comment (snrssc) [inline java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java:107]: Printing JDBC URL in logs printing password also in logs. May be we can remove password and print or simply ignore printing jdbc url for now. It's okay to handle this in a new issue too.

**Review by snrssc — COMMENTED** [review]
Except the logic making partition read mandatory, everything else looks good.

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java (hunk 1 above)
Comment (shashank-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java:83]: No. that is not correct.
It is a multiline condition If (concatedPartitionProps != blank && (jdbcSQLPartitionColumn != blank || ...)) then trow exception.

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java (hunk 2 above)
Comment (shashank-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java:107]: Good point thanks. Removed it.

**Pull request merged**